src/cript/api/api.py

```diff
@@ -484,6 +502,404 @@ def _check_initial_host_connection(self) -> None:
         except Exception as exc:
             raise CRIPTConnectionError(self.host, self._api_token) from exc
 
+    # ================ helpers ==============
+    @classmethod
+    def object_exists(cls, node: str, name: str):
+        host = Config.host
+        token = Config.api_token
+
+        # FOR NOW : hard code brilis proj name
+        # object_name = "P2A3DAA" # not used
+
+        api_url = f"{host}/api/v1/search/exact/{node}/?q={name}"  # Ensure this URL lists all objects of the type
+        headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
+
+        try:
+            response = requests.get(api_url, headers=headers)
+
+            data = response.json()["data"]
+            print("\nlength should be 1: ", len(data.get("result")))
+
+            if len(data.get("result")) == 1:
+                item = data.get("result")[0]
+
+                return True, item.get("uuid")
+
+            return False, None
+
+        except requests.RequestException as e:
+            print(f"An error occurred: {e}")
+            return False, None
+
+    @staticmethod
+    def fetch_object_data(object_type: str, uuid: str):
```

Comment:
should be through paginator and search.

Reply:
I realize, that this comment is not helpful at the moment, because I am refactoring the paginator.
but eventually, we should strive to not duplicate code paths that essentially do the same thing....

Reply:
@InnocentBug  could you give a quick example where the call to the paginator would look like to retrieve an already existing node?

Reply:
Sure.
```python
import cript

with cript.API() as cript_api: # Assumes you have the host and token in the environment.
    # If you know its uuid
    uuid_paginator = cript_api.search(node_type=cript.Project, search_mode=cript.SearchModes.UUID, value_to_search="e7381525-64b9-4849-a416-57aa1e09044f")
    project = next(uuid_paginator)
    print(project)

    # if you know its name
    exact_name_paginator = cript_api.search(node_type=cript.Material, search_mode=cript.SearchModes.EXACT_NAME, value_to_search="Sodium polystyrene sulfonate")
    material = next(exact_name_paginator)
    print(material)
```